File: src/test/java/kitchenpos/application/fake/FakeOrderTableRepository.java
package kitchenpos.application.fake;

import kitchenpos.domain.OrderTable;
import kitchenpos.domain.OrderTableRepository;

import java.util.*;

public class FakeOrderTableRepository implements OrderTableRepository {
    Map<UUID,OrderTable> memory = new HashMap<>();
    @Override
    public Optional<OrderTable> findById(UUID orderTableId) {
        return Optional.ofNullable(memory.get(orderTableId));
    }

    @Override
    public OrderTable save(OrderTable orderTable) {
        memory.put(orderTable.getId(), orderTable);
        return memory.get(orderTable.getId());
    }

    @Override
    public List<OrderTable> findAll() {
        return new ArrayList<>(memory.values());
    }
}

File: src/test/kotlin/kitchenpos/application/fake/FakeOrderTableRepository.kt
package kitchenpos.application.fake

import kitchenpos.domain.OrderTable
import kitchenpos.domain.OrderTableRepository
import java.util.*

class FakeOrderTableRepository : OrderTableRepository {
    private val orderTables = mutableMapOf<UUID, OrderTable>()

    override fun save(entity: OrderTable?): OrderTable {
        orderTables[entity!!.id] = entity
        return entity
    }

    override fun findById(id: UUID?): Optional<OrderTable> {
        return Optional.ofNullable(orderTables[id])
    }

    override fun findAll(): MutableList<OrderTable> {
        return orderTables.values.toMutableList()
    }
}
